pkg/services/control/server/object_revive.go

```diff
@@ -0,0 +1,52 @@
+package control
+
+import (
+	"context"
+
+	"github.com/nspcc-dev/neofs-node/pkg/services/control"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (s *Server) ReviveObject(_ context.Context, request *control.ReviveObjectRequest) (*control.ReviveObjectResponse, error) {
+	err := s.isValidRequest(request)
+	if err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	// check availability
+	err = s.ready()
+	if err != nil {
+		return nil, err
+	}
+
+	var addr oid.Address
+	err = addr.DecodeString(request.GetBody().GetObjectAddress())
```

Comment:
Why a string representation here? I'd not expect this in a protobuf protocol, you can avoid decoding.

Reply:
I don't really understand. Do I need to transmit via bytes? I did the same as the [status](https://github.com/nspcc-dev/neofs-node/blob/5c5c42c1c275eb40c33fdd80869b10f07c1d7779/pkg/services/control/server/object_status.go#L27).

Reply:
Whoa. Really unexpected. Well, it's a good reference, but I'd say it's a bug of `status` implementation. We can either divert here and make it better (it simplifies things) or follow established practice. Not critical then.

Reply:
i thought i was expired by some another example for `status` but i do not find any string implementation (it is strange, cause i am for raw bytes too) so let it be raw bytes then cause we also have it in control API: https://github.com/nspcc-dev/neofs-node/blob/5c5c42c1c275eb40c33fdd80869b10f07c1d7779/pkg/services/control/service.proto#L117-L119

status can be fixed separately

Reply:
Can you please explain where I should use `bytes` and where `string`?
For example, here:
https://github.com/nspcc-dev/neofs-node/blob/1c786839073c7e4a08ab753d3d6a5e98d8d8be7c/pkg/services/control/service.proto#L454-L456
I see that `shard_id` can also be `bytes`, as shown here:
https://github.com/nspcc-dev/neofs-node/blob/f858a81cbd1c9116042c0e92adad72405c8741d7/pkg/services/control/types.proto#L123

Reply:
i think shard ID is ok to be `string` cause there is no fixed public rule for bytes -> string conversion (for addresses we know it), but for things from our public API it is ok to be bytes and cli can convert is inside according to SDK
@roman-khimov

Reply:
Shard ID can be a string, it's deeply internal to the node, no one knows how to convert it properly into something human-oriented.